vm-setup/roles/v1aX_integration_test/templates/cluster-template-cluster.yaml

```
@@ -1,4 +1,4 @@
-apiVersion: cluster.x-k8s.io/v1alpha3
+apiVersion: cluster.x-k8s.io/${ CAPI_VERSION }
```

This was confusing me at first but now I understand why we have both bash and jinja2 syntax in the same file!
For anyone struggling with the same, it is because we first template this with Ansible, and then with `clusterctl conig cluster`. See [this](https://github.com/metal3-io/metal3-dev-env/blob/468a8a1cd0822e3d99baaa964b31f574cbb8bcb9/vm-setup/roles/v1aX_integration_test/tasks/generate_templates.yml#L13-L34).

I think we should use jinja2 templates when possible instead of bash since it gives more/easier control over formatting and is the more "natural" templating system when working with Ansible.
```suggestion
apiVersion: cluster.x-k8s.io/{{ CAPI_VERSION }}
```

---

I was thinking the same but then left it for a separate cleanup

---

Done